src/js/overlay.js

```diff
@@ -225,16 +219,33 @@ Overlay.prototype.show = function() {
 
 	this.closeOnExternalClick = this.closeOnExternalClick.bind(this);
 	this.delegates.doc.on('click', 'body', this.closeOnExternalClick);
+	this.delegates.doc.on('touchstart', 'body', this.closeOnExternalClick);
```

Comment:
touchstart seems like a rather naive way to detect if a user wanted to touch outside the modal box. I'm sure it's fine as a first iteration, but any thoughts on how we can improve on that? /cc @pornel